### PR TITLE
fix($controller): allow identifiers containing `$`

### PR DESCRIPTION
**Pull request by gkalpak**
As discussed in https://github.com/angular/angular.js/issues/13664#issuecomment-170536024.

### CODE REVIEW

File: src/ng/controller.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 var $controllerMinErr = minErr('$controller');
 
 
-var CNTRL_REG = /^(\S+)(\s+as\s+(\w+))?$/;
+var CNTRL_REG = /^(\S+)(\s+as\s+([\w$]+))?$/;
```
Comment (petebacondarwin) [inline src/ng/controller.js]: Perhaps only allow `$` at the beginning or is that being too strict?

Comment (gkalpak) [inline src/ng/controller.js]: Yeah, it crossed my mind, but I thought it was too strict. It complicates things unnecessarily, I think.
(Note that it's already possible to use many more characters using `controllerAs`.)

**Comment by gkalpak** [comment]
@petebacondarwin, do you think we should backport it to `v1.4.x` ?

**Comment by petebacondarwin** [comment]
Yep

**Comment by gkalpak** [comment]
Backported to `v1.4.x` as 2563ff7.

**Pull request closed without merging**